### PR TITLE
TST: Test pd.Grouper base floating point error (#25161)

### LINKED ISSUE #25161: Groupby creates emptu groups depending on base parameter
#### Code Sample, a copy-pastable example if possible

```python
"Generate test dataframe"
case = 1
if case == 1:
    start = '2018-11-26 16:17:43.510000'
else:
    start = '2018-11-26 16:17:43.500000'

rng = pd.date_range(start, periods=10, freq='1S')
df = pd.DataFrame({'a':np.random.randn(len(rng)),'b':np.random.randn(len(rng))}, index=rng)

"Set interval and start time of the buckets"
interval = dt.timedelta(minutes=10)
t0 = df.index[0]
base = t0.minute + (t0.second +t0.microsecond/1e6)/60
groups = df.groupby(pd.Grouper(freq=interval, base=base))

print(groups.size())

```
#### Problem description

The code above generates either 6 or 7 groups depending if the dataframe starts at  '2018-11-26 16:17:43.500000' (case 1) or  '2018-11-26 16:17:43.510000' (case 2).

The correct output is clearly the one obtained in case 2. Case 1, instead, creates an empty group at the end of the dataframe. This can cause troubles with groupby.apply() if the applied function does not handle empty dataframes. 

#### Actual Output
2018-11-26 16:17:43.510    10
**2018-11-26 16:27:43.510      #0**
dtype: int64

#### Expected Output
2018-11-26 16:17:43.510    10
dtype: int64
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.1.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None
 
pandas: 0.23.4
pytest: 4.0.2
pip: 18.1
setuptools: 40.6.3
Cython: 0.29.2
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: 1.8.2
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 3.0.2
openpyxl: 2.5.12
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.2
lxml: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by ihsansecer**
- [x] closes #25161
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/groupby/test_grouping.py [hunk 1]
```diff
@@ -203,6 +203,18 @@ def test_grouper_creation_bug(self):
         expected = s.groupby(level='one').sum()
         assert_series_equal(result, expected)
 
+        # GH25161
+        s = Series(np.arange(10),
+                   index=pd.date_range(
+                       "2018-11-26 16:17:43.51",
+                       periods=10, freq="1S"))
+        t0 = s.index[0]
+        base = t0.minute + (t0.second + t0.microsecond / 1e6) / 60
```
Comment (WillAyd) [inline pandas/tests/groupby/test_grouping.py]: Can you replace expressions like these with literal values? I realize this is copied directly from referenced issue but writing it this way makes the test more difficult to reason about than it needs to be

**Review by WillAyd — CHANGES_REQUESTED** [review]
Can you create this as a separate test? The existing one you are appending to is rather large as is

File: pandas/tests/groupby/test_grouping.py (hunk 1 above)
Comment (ihsansecer) [reply pandas/tests/groupby/test_grouping.py]: Done! Thanks

File: pandas/tests/groupby/test_grouping.py [hunk 2]
```diff
@@ -303,6 +303,18 @@ def test_grouper_getting_correct_binner(self):
     def test_grouper_iter(self, df):
         assert sorted(df.groupby('A').grouper) == ['bar', 'foo']
 
+    def test_grouper_base(self):
```
Comment (jreback) [inline pandas/tests/groupby/test_grouping.py]: can you move to test pandas/tests/resample/test_timegrouper and also test in .resample (which is what this calls)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_grouping.py (hunk 2 above)
Comment (ihsansecer) [reply pandas/tests/groupby/test_grouping.py]: Changed it

File: pandas/tests/resample/test_datetime_index.py [hunk 3]
```diff
@@ -751,6 +751,19 @@ def test_resample_base():
...
+    # GH25161
+    dt = pd.to_datetime(["2018-11-26 16:17:43.51",
+                         "2018-11-26 16:17:44.51",
+                         "2018-11-26 16:17:45.51"])
+    s = Series(np.arange(3), index=dt)
+
+    base = 17 + 43.51 / 60
+    resampled = s.resample("3min", base=base).mean()
+    exp_index = DatetimeIndex(["2018-11-26 16:17:43.51"])
+    tm.assert_index_equal(resampled.index, exp_index)
```
Comment (gfyoung) [inline pandas/tests/resample/test_datetime_index.py]: FWIW, I would prefer to just check the whole `Series`, not the index.

Comment (ihsansecer) [reply pandas/tests/resample/test_datetime_index.py]: I see. Will change it today

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ihsansecer

**Pull request merged**